### PR TITLE
feat: add Odin language support

### PR DESCRIPTION
**Pull request by laytan**
https://odin-lang.org

### CODE REVIEW

**Comment by amaanq** [comment]
How coincidental! I just completed my own Odin parser today and was planning to PR it soon (I was not a fan of the "original" one, the one you forked) - taking a look at yours I noticed about ~440 files in the Odin core repo fail so I'd like to use mine instead

File: queries/odin/indents.scm [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ (foreign_block)
+ (block)
+ (top_block)
+ (compound_literal)
+ (proc_call)
+ (assignment_statement)
+ (const_declaration)
+ (var_declaration)
+ (switch_statement)
+] @indent
```
Comment (lucario387) [inline queries/odin/indents.scm:11]: This is using the old syntax
The equivalents are
- `@indent` -> `@indent.begin`
- `@indent_end` -> `@indent.end`
- `@branch` -> `@indent.branch`

File: queries/odin/indents.scm [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+[
+ ")"
+ "}"
+] @indent_end @branch
+
```
Comment (lucario387) [inline queries/odin/indents.scm:17]: This is using the old syntax
The equivalents are

- `@indent` -> `@indent.begin`
- `@indent_end` -> `@indent.end`
- `@branch` -> `@indent.branch`

**Comment by laytan** [comment]
@amaanq The grammar currently fails to parse the return types correctly, for example `-> (string, bool)` it for some reason tries to parse it as a group expression, and then errors when it doesn't know what to do with the `->`.

Not sure why, but I think when I figure that out that there are no more errors, I will get to that today hopefully. I will also double check all the core files, that's a good idea!

**Comment by laytan** [comment]
Closing in favour of #4784

**Pull request closed without merging**